Fase-1/API/self-order-management/schema.sql

```diff
@@ -35,19 +38,30 @@ CREATE TABLE IF NOT EXISTS customer
 
 CREATE TABLE IF NOT EXISTS "order"
 (
-    order_id SERIAL PRIMARY KEY,
-    order_total NUMERIC(15,2) NOT NULL,
-    order_status VARCHAR(255) NOT NULL,
-    order_customer_document VARCHAR(20),
+    order_number SERIAL PRIMARY KEY,
+    order_date DATE NOT NULL,
     order_customer_nickname VARCHAR(255) NOT NULL,
+    order_customer_document VARCHAR(20),
+    order_status VARCHAR(255) NOT NULL,
+    order_total NUMERIC(15,2) NOT NULL,
     CONSTRAINT fk_order_customer_id FOREIGN KEY(order_customer_document) REFERENCES customer(customer_document)
 );
 
 CREATE TABLE IF NOT EXISTS order_item
 (
     order_item_id SERIAL PRIMARY KEY,
-    order_item_item_name VARCHAR(255) NOT NULL,
-    order_item_order_id SERIAL NOT NULL,
-    CONSTRAINT fk_order_item_item_id FOREIGN KEY(order_item_item_name) REFERENCES item(item_name),
-    CONSTRAINT fk_order_item_order_id FOREIGN KEY(order_item_order_id) REFERENCES "order"(order_id)
+    order_item_product_number INTEGER NOT NULL,
+    order_item_order_number SERIAL NOT NULL,
+    CONSTRAINT fk_order_item_product_id FOREIGN KEY(order_item_product_number) REFERENCES product(product_number),
+    CONSTRAINT fk_order_item_order_id FOREIGN KEY(order_item_order_number) REFERENCES "order"(order_number)
 );
+
+CREATE TABLE IF NOT EXISTS payment
+(
+    payment_order_number SERIAL PRIMARY KEY,
+    payment_external_id UUID NOT NULL,
+    payment_created_at TIMESTAMP NOT NULL,
+    payment_status TEXT NOT NULL,
+    payment_status_changed_at TIMESTAMP NOT NULL,
+    CONSTRAINT fk_payment_order_id FOREIGN KEY(payment_order_number) REFERENCES "order"(order_number)
+)
```

Comment:
Pagamentos de pedidos (`order_number`) são executados por um provedor externo, que por contrato identificam o pagamento com um UUID (`external_id`). Essa é uma relação simples, 1-to-1, e não mantêm histórico de transações que falharam ou completaram. Motivação: nosso domínio não é pagamento, evidente pelo uso de um terceiro, que por sua vez está habilitado para fornecer todos os dados, auditoria, e insights em relação aos pagamentos. Ainda assim precisamos manter a informação mínima de pagamento dos pedidos.